analytics/tracker.go

```diff
@@ -0,0 +1,135 @@
+package analytics
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/bitrise-io/bitrise/configs"
+	"github.com/bitrise-io/bitrise/models"
+	"github.com/bitrise-io/bitrise/version"
+	"github.com/bitrise-io/go-utils/pointers"
+	"github.com/bitrise-io/go-utils/v2/analytics"
+	"github.com/bitrise-io/go-utils/v2/env"
+	stepmanModels "github.com/bitrise-io/stepman/models"
+	log "github.com/sirupsen/logrus"
+)
+
+// Tracker ...
+type Tracker interface {
+	SendWorkflowStarted(properties analytics.Properties, title string)
+	SendWorkflowFinished(properties analytics.Properties, failed bool)
+	SendStepStartedEvent(properties analytics.Properties, infoModel stepmanModels.StepInfoModel, inputs map[string]interface{})
+	SendStepFinishedEvent(properties analytics.Properties, results models.StepRunResultsModel)
+	Wait()
+}
+
+type tracker struct {
+	tracker       analytics.Tracker
+	envRepository env.Repository
+}
+
+// NewTracker ...
+func NewTracker() Tracker {
+	if isAnalyticsDisabled() {
+		return NoOpTracker{}
+	}
+	return tracker{
+		tracker:       analytics.NewDefaultTracker(),
+		envRepository: env.NewRepository(),
+	}
+}
+
+// SendWorkflowStarted ...
+func (t tracker) SendWorkflowStarted(properties analytics.Properties, title string) {
+	isCI := t.envRepository.Get(configs.CIModeEnvKey) == "true"
+	isPR := t.envRepository.Get(configs.PRModeEnvKey) == "true"
+	isDebug := t.envRepository.Get(configs.DebugModeEnvKey) == "true"
+	isSecretFiltering := t.envRepository.Get(configs.IsSecretFilteringKey) == "true"
+	isSecretEnvsFiltering := t.envRepository.Get(configs.IsSecretEnvsFilteringKey) == "true"
+	stateProperties := analytics.Properties{
+		"workflow_name":         title,
+		"ci_mode":               isCI,
+		"pr_mode":               isPR,
+		"debug_mode":            isDebug,
+		"secret_filtering":      isSecretFiltering,
+		"secret_envs_filtering": isSecretEnvsFiltering,
+	}
+	stateProperties.AppendIfNotEmpty("build_slug", t.envRepository.Get("BITRISE_BUILD_SLUG"))
+	stateProperties.AppendIfNotEmpty("parent_step_unique_id", t.envRepository.Get("BITRISE_STEP_UNIQUE_ID"))
+	stateProperties.AppendIfNotEmpty("log_level", t.envRepository.Get(configs.LogLevelEnvKey))
```

Comment:
What is the value in collecting the log level of builds? I have a feeling that almost no one knows about setting the log level on the VM (compared to local Bitrise runs). Also, the last time I tried the debug level, it was unusable thanks to the noisy logging.

Reply:
We'll be able to answer the question: `Who sets the log level from the default?` I'm not sure if it is possible or if we ever want to answer that. I'll just remove it for now, we can add back if it will be needed.